src/main/java/edu/hm/hafner/analysis/parser/IssuesFilter.java

```diff
@@ -0,0 +1,188 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.util.regex.Pattern;
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.Issues;
+
+public class IssuesFilter {
+    private boolean includeFilterSet;
+    private boolean excludeFilterSet;
+    private Issues filteredIssues;
+    private Pattern patternIncludeFilename;
+    private Pattern patternIncludeCategory;
+    private Pattern patternIncludePriority;
+    private Pattern patternIncludeMessage;
+    private Pattern patternIncludeDescription;
+    private Pattern patternIncludeType;
+    private Pattern patternIncludeColumnStart;
+    private Pattern patternIncludeColumnEnd;
+    private Pattern patternExcludeFilename;
+    private Pattern patternExcludeCategory;
+    private Pattern patternExcludePriority;
+    private Pattern patternExcludeMessage;
+    private Pattern patternExcludeDescription;
+    private Pattern patternExcludeType;
+    private Pattern patternExcludeColumnStart;
+    private Pattern patternExcludeColumnEnd;
+
+
+    public IssuesFilter(Issues issues){
+        includeFilterSet = false;
```

Comment:
Wenn du die Issues im Konstruktor übergibst, kannst du den Filter nur einmal verwenden. Hast du dir überlegt die issues in der filter Methode zu übergeben?